sherpa/astro/fake.py

```diff
@@ -134,62 +181,21 @@ def fake_pha(data, model,
     elif not callable(method):
         raise ArgumentTypeErr("badarg", "method", "a callable")
 
-    if is_source:
-        model = get_response_for_pha(data, model, bkg_srcs={},
-                                     pileup_model=pileup_model, id=id)
-
-    # Get one RMF. Hopefully all of them have the same number of
-    # channels, but that sanity check should really be done elsewhere.
-    rmf0 = data.get_rmf(data.response_ids[0])
-    data.channel = np.arange(1, rmf0.detchans + 1)
-
-    # Calculate the source model, and take a Poisson draw based on
-    # the source model.  That becomes the simulated data.
-    data.counts = method(data.eval_model(model))
-
-    # Add in background counts:
-    #  -- Scale each background properly given data's
-    #     exposure time, BACKSCAL and AREASCAL
-    #  -- Take average of scaled backgrounds
-    #  -- Take a Poisson draw based on the average scaled background
-    #  -- Add that to the simulated data counts
-    #
-    # Adding background counts is OPTIONAL, only done if user sets
-    # "bkg" argument to fake_pha.  The reason is that the user could
-    # well set a "source" model that does include a background
-    # component.  In that case users should have the option to simulate
-    # WITHOUT background counts being added in.
+    # Evaluate the model. This assumes the model term contains a
+    # response and any scaled background components.
     #
-    if add_bkgs:
-        if bkg_models is None:
-            bkg_models = {}
-
-        nbkg = len(data.background_ids)
-        b = 0
+    model_prediction = data.eval_model(model)
+
+    if include_bkg_data:
+        # If there are background components then we sum up the data,
+        # after scaling to match the source data set, and use that as a
+        # source term. Note that get_background_scale will correct for
```

Comment:
It's amazing how someone who understand Sherpa's internals can clean up my crappy code.

Reply:
I'm not convinced the existing documentation describes (or, at the time this code was written, described) the assumptions that various operations require or make, so it can be hard to work out what the current "best approach" is.